activesupport/test/core_ext/range_ext_test.rb

```diff
@@ -253,13 +253,6 @@ def test_each_on_time_with_zone
     end
   end
 
-  def test_step_on_time_with_zone
-    twz = ActiveSupport::TimeWithZone.new(nil, ActiveSupport::TimeZone["Eastern Time (US & Canada)"], Time.utc(2006, 11, 28, 10, 30))
-    assert_raises TypeError do
-      ((twz - 1.hour)..twz).step(1) { }
-    end
```

Comment:
I think we should keep that test (or a similar one) but with a ruby version check. `if RUBY_VERSION < "3.4"`, and then add another test to make sure it works as expected on 3.4+